### ISSUE TITLE
Dropout arguments isn't working

### ISSUE THREAD

**Issue by PotatoPeeZy** [ultralytics/ultralytics#15652]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train, Val, Other

### Bug

I wanted to train a custom classification model with YOLOv8x-cls model. But it seems like it is overfitting. So I tried to introduce dropout to the model. But it seems like the argument "dropout" isn't working because I passed 0.2, 0.4, 0.6, or even 1 in the argument. But the results are more or less same in every training run. I've included the code block I used to train the model. 

### Environment

Ultralytics YOLOv8.2.56  Python-3.11.9 torch-2.3.1 CUDA:0 (NVIDIA GeForce RTX 4090, 24564MiB)
Setup complete  (12 CPUs, 24.0 GB RAM, 247.0/499.1 GB disk)

### Minimal Reproducible Example

```
from ultralytics import YOLO
model1 = YOLO("yolov8x-cls.pt")
model1.train(data='C:\\Users\\CSE-AI-Lab\\Desktop\\Shishir Workplace\\Sundori Leaf Data Model\\Leaf Data\\40 class leaf only Data 224 splitted\\fold_1', dropout = 0.4 ,epochs=10, cache = False, batch = -1, imgsz = 224, device = 0,name ="40% dropout")
```

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

**Comment by Y-T-G** [comment]
For dropout to work, you will need to add `nn.Dropout` layers to the `yaml`.

https://github.com/ultralytics/ultralytics/tree/main/ultralytics/cfg/models/v8/yolov8-cls.yaml

Like:
`[-1, 1, nn.Dropout, [0.5]]`

**Comment by Y-T-G** [comment]
@Jordan-Pierce You can add `Dropout` layers to any model. But the `dropout` argument would only work with classification models. The rest would just use the default probability you set in the layer.
